packages/env/test/index.test.ts

```diff
@@ -263,4 +264,30 @@ _:b1 sh:path schema:name .
       })
     }
   })
+
+  describe('dataset factory ext', () => {
+    it('has toStream utility', async () => {
+      // given
+      const quad = $rdf.quad($rdf.namedNode('foo'), $rdf.namedNode('bar'), $rdf.namedNode('baz'))
+      const dataset = env.dataset([quad])
+
+      // when
+      const streamedQuads = await getStreamAsArray(env.dataset.toStream(dataset))
+
+      // then
+      expect(streamedQuads).to.deep.contain.members([quad])
+    })
+
+    it('has fromStream utility', async () => {
+      // given
+      const quad = $rdf.quad($rdf.namedNode('foo'), $rdf.namedNode('bar'), $rdf.namedNode('baz'))
+      const dataset = env.dataset([quad]).toStream()
+
+      // when
+      const streamedQuads = await env.dataset.fromStream(dataset)
+
+      // then
+      expect([...streamedQuads]).to.deep.contain.members([quad])
```

Comment:
what about throwing in `toCanonical` to assert the roundtrip? any dataset and the one obtained via `toStream/fromStream` have the same canonical form.